lesson-1-7-demo/src/DinnerParty.jsx

```diff
@@ -26,54 +26,104 @@ const attendDinnerParty = ({
         reject({
           id: ++id,
           status: "🫣 + 🤢",
-          attended: false,
           timestamp: new Date(),
         });
       } else if (isSick && !isFlaky) {
         reject({
           id: ++id,
           status: "🤢",
-          attended: false,
           timestamp: new Date(),
         });
       } else if (!isSick && isFlaky) {
         reject({
           id: ++id,
           status: "🫣",
-          attended: false,
           timestamp: new Date(),
         });
       } else {
         resolve({
           id: ++id,
           status: "🥳",
-          attended: true,
           timeestamp: new Date(),
         });
       }
     }, timeout);
   });
 
+// DINNER_PARTY_INVITE_RECEIVED
+// DINNER_PARTY_INVITE_ACCEPTED
+// DINNER_PARTY_INVITE_REJECTED
+// HEALTH_CHANGED
+// FRIENDSHIP_LEVEL_CHANGED
+
+const initialReducerState = {
+  health: 0.5,
+  friendshipLevel: FRIENDSHIP_LEVELS.FRIEND.value,
+  isDeciding: false,
+  dinnerPartyResults: [],
+};
+
 const DinnerParty = () => {
   // TODO: Add state to manage form
+  const [health, setHealth] = useState(0.5);
+  const [friendshipLevel, setFriendshipLevel] = useState(
+    FRIENDSHIP_LEVELS.FRIEND.value
+  );
+  const [isDeciding, setIsDeciding] = useState(false);
+  const [dinnerPartyResults, setDinnerPartyResults] = useState([]);
+  const handleHealthChange = (event) => {
+    setHealth(Number(event.target.value));
+  };
+  const handleFriendshipLevelChange = (event) => {
+    setFriendshipLevel(Number(event.target.value));
+  };
+
+  const handleDinnerPartyInvite = () => {
+    console.log("Handling dinner party invite");
+    console.log({ health, friendshipLevel });
+    setIsDeciding(true);
+    // async request! how do we get access to the results?
+    attendDinnerParty({
```

Comment:
this is an async request so we must use `.then()` and `.catch()` to get the results of the request